src/modules/editor/components/extensions/helpers/MediumDialog.tsx

```diff
@@ -0,0 +1,215 @@
+import * as React from "react";
+
+import { client } from "../../../../apolloClient";
+import { queryAccountIDs, IMedium } from "../../../queryHelpers";
+
+import { Dialog, DialogActions, DialogTitle, DialogContent } from '@rmwc/dialog';
+import { TextField } from '@rmwc/textfield';
+import { Select } from "@rmwc/select";
+import { Button } from '@rmwc/button';
+import { CircularProgress } from '@rmwc/circular-progress';
+
+import { createUseStyles } from "react-jss";
+
+import { snackbarQueue } from '../../../../snackbarQueue';
+import { ContributorsField } from '../../helpers/ContributorsField';
+import { CREATE_MEDIUM_MUTATION } from '../../../queryHelpers';
+import classes from "*.module.css";
+
+
+interface IVariables {
+    title: string,
+    user_id: number,
+    caption?: string,
+    attachment_b64: string,
+    media_type: string,
+    is_featured: boolean
+}
+
+interface IData {
+    id: string
+}
+
+const useStyles = createUseStyles({
+    DialogForm: {
+        display: "flex",
+        flexDirection: "column"
+    },
+    Input: {
+        marginTop: "10px"
+    },
+    UploadButton: {
+        display: "flex",
+        flexDirection: "row"
+    }
+})
+
+interface IProps {
+    open: boolean,
+    onClose: (medium?: IMedium) => any,
+    initialData?: IVariables,
+    isFeatured: boolean
+}
+
+export const MediumDialog: React.FC<IProps> = ({ open, onClose, initialData, isFeatured }) => {
```

Comment:
This component feels a little complicated. Plus is it a `MediumDialog` if it also includes a form?

Reply:
It is complicated, but not sure how I'd make it better. It's all one problem. I'd say it is a dialog as although the meat is a form, it's displayed as a dialog, and that's consistent with my naming scheme in the project